charts/opensearch/templates/statefulset.yaml

```diff
@@ -334,6 +338,10 @@ spec:
           value: "{{ .Values.opensearchJavaOpts }}"
         - name: node.roles
           value: "{{ template "opensearch.roles" . }}"
+        {{- if .Values.singleNode }}
+        - name: discovery.type
+          value: "single-node"
+        {{- end }}
```

Comment:
Maybe better as: 

```suggestion
          value: {{ if .Values.singleNode }}single-node{{ else }}multi-node{{ end }}
```

Thoughts?

Reply:
Wait, I didn't think to check as I don't have any experience with the opensearch code base, but are you sure there is a `multi-node` discovery type setting?

The sources I've found only list the `zen` type: https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/discovery/DiscoveryModule.java#L84

Reply:
Wouldn't it be best to just omit the env var if we don't set it to `single-node`?
And not make any assumptions about other discovery types in the helm chart?

Reply:
https://www.elastic.co/guide/en/elasticsearch/reference/current/modules-discovery-settings.html#modules-discovery-settings

Yeah you might be right, maybe this didn't make it over to Opensearch from ES. Weird.

Reply:
Weird indeed. Do you think it would be fine to keep the `if` as it is then? Otherwise one would have to update the chart should the default `discovery.type` ever change.

Reply:
Looks to me like we need `discovery.type` only if we want `single-node`
https://opensearch.org/docs/latest/opensearch/install/docker/#run-the-image